### PR TITLE
Added featured and example tags to packages, updated styles, fixed admin template bug

### PR DESCRIPTION
**Pull request by pennyfx**
- Added icons for search meta data(author,forks,activity,size) on search page
- Fixed a template bug related to Jinja migration
- Added example and featured flags to packages.  bugs 695530, 695536
- Created an admin page that allows you to search by package id and toggle featured and example flags
- This pull requires schematic migration

### CODE REVIEW

File: apps/base/templates/monitor.html [hunk 1]
```diff
@@ -29,8 +29,8 @@
 
 <h2 class="UI_Heading" style="padding-top: 3em">Free space</h2>
 <ul>
-  {% for path, size in free.items() %}
-  <li>{{ path }}:  {{ size }}kB</li>
+  {% for path, size in free %}
```
Comment (seanmonstar) [inline apps/base/templates/monitor.html:32]: is `free` a dict, or a tuple?

File: apps/base/views.py [hunk 2]
```diff
@@ -130,10 +131,10 @@ def monitor(request):
     s_path = '%s/' % settings.SDKDIR_PREFIX
     x = os.statvfs(x_path)
     s = os.statvfs(s_path)
-    data['free'] = {
-            'xpi_targetdir %s' % x_path: (x.f_bavail * x.f_frsize) / 1024,
-            'sdkdir_prefix %s' % s_path: (s.f_bavail * s.f_frsize) / 1024
-            }
+    data['free'] = [
```
Comment (seanmonstar) [inline apps/base/views.py:134]: ahh i see, this answers my question above. why the change from a dict?

File: apps/base/views.py [hunk 3]
```diff
@@ -190,13 +191,31 @@ def monitor(request):
...
+def get_package(request):    
+    package = get_object_or_404(Package, pk=request.GET['package_id'])    
+    return render_to_response('admin/_package_result.html', {
+            'package': package
+        }, context_instance=RequestContext(request))
+
+@user_passes_test(lambda u: u.is_superuser)
+def update_package(request):
+    package = get_object_or_404(Package, pk=request.POST['package_id'])  
+    if request.POST.__contains__('featured'):
```
Comment (seanmonstar) [inline apps/base/views.py]: best practices for checking if a dict contains a key is to use `in`.

`if 'featured' in request.POST`

File: apps/search/templates/_package_result.html [hunk 4]
```diff
@@ -2,21 +2,26 @@
...
+            <span class="featured" title="Featured">featured</span>
+        {% endif %}
+        {% if package.example %}
+            <span class="example" title="Example">example</span>
+        {% endif %}
     </h3>
     <p class="description">{{ package.description }}</p>
     <ul class="search_meta">
-        <li><span>author:</span>
+        <li title="Author" class='author'><span>&nbsp;</span>
```
Comment (seanmonstar) [inline apps/search/templates/_package_result.html]: nitpicks: use " for HTML attributes.

File: apps/base/views.py (hunk 2 above)
Comment (pennyfx) [inline apps/base/views.py:134]: An array of tuples seemed more appropriate.

If it was something like 

```
free = { 'xpi_targetdir' : { 'path': x_path, 'available':  x.f_bavail * x.f_frsize } , ... } 
```

then it would make more sense to be a dictionary. In my opinion...

I guess in my mind, I think of dictionaries as having easy and known keys for ease of access later on.  In this scenario, it was just iterating over a list of 2 values.

File: apps/base/templates/monitor.html (hunk 1 above)
Comment (pennyfx) [inline apps/base/templates/monitor.html:32]: array of tuples

File: media/base/js/Admin.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                     if (callback) callback();
+                },
+                onFailure: function(err){
+                    if(err.status == 404){
+                        $('package_item').set('html', '');
+                    }
+                    setMsg(err.status + " " + err.statusText);
+                }
+            });
+        r.send('package_id='+id+'&'+field+'='+!enabled);
```
Comment (seanmonstar) [inline media/base/js/Admin.js:42]: Nothing wrong with sending it like this, but I personally find it easier to use an object on the `data` option.

``` js
var data = {
    package_id: id
};
data[field] = !enabled;
new Request({
    data: data
}).send();
```

File: media/search/css/Search.css [hunk 6]
```diff
@@ -192,12 +193,53 @@ body {
...
     
-    #SearchResults .package ul.search_meta li span{
-        color:#666;
+    #SearchResults .package ul.search_meta li span{        
+        opacity: 0.35;
+        padding-right: 10px;
+    }
+    
+    #SearchResults .package ul.search_meta li.size span{
+        background-image:url('../img/size.png');
```
Comment (seanmonstar) [inline media/search/css/Search.css:206]: yay icons!

File: media/base/js/Admin.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
+window.addEvent('domready', function(e){
+    var setMsg = function(msg){
+        $('package_msg').set('html', msg);
```
Comment (seanmonstar) [inline media/base/js/Admin.js]: Unless you expect the element to be destroy and re-created, it's best to do the query only once, and save the element in a local variable.

File: apps/base/views.py (hunk 2 above)
Comment (seanmonstar) [inline apps/base/views.py:134]: looks good to me

File: apps/base/views.py [hunk 8]
```diff
@@ -190,13 +191,31 @@ def monitor(request):
...
+            'package': package
+        }, context_instance=RequestContext(request))
+
+@user_passes_test(lambda u: u.is_superuser)
+def update_package(request):
+    package = get_object_or_404(Package, pk=request.POST['package_id'])  
+    if request.POST.__contains__('featured'):        
+        package.featured = request.POST.get('featured') == 'true'
+        
+    if request.POST.__contains__('example'):
```
Comment (seanmonstar) [inline apps/base/views.py]: Use `in` here as well.

**Comment by seanmonstar** [comment]
While the amount of data posted is small, it's not a bad habit to use `Forms`. (AMO uses them extensively.)

As I touch various views, I try to convert them to using a Form when I can.

File: apps/base/views.py (hunk 3 above)
Comment (pennyfx) [inline apps/base/views.py]: That's way better than the double underscore business. thanks

Comment (seanmonstar) [inline apps/base/views.py]: Actually, the double-underscored method is a magic method, and if a class declares it, then the `in` operator will work on it.
